multy_core/src/ethereum/ethereum_stream.cpp

```diff
@@ -0,0 +1,122 @@
+/* Copyright 2018 by Multy.io
+ * Licensed under Multy.io license.
+ *
+ * See LICENSE for details
+ */
+
+#include "multy_core/src/ethereum/ethereum_stream.h"
+
+#include "multy_core/src/ethereum/ethereum_extra_data.h"
+
+#include "multy_core/src/exception_stream.h"
+#include "multy_core/src/utility.h"
+#include "multy_core/binary_data.h"
+
+#include <array>
+#include <unordered_map>
+#include <vector>
+
+namespace
+{
+using namespace multy_core::internal;
+
+const size_t ETHEREUM_SIZE_VARIABLE_FUNCTION_CONTRACT = 32;
+static const std::array<const uint8_t, ETHEREUM_SIZE_VARIABLE_FUNCTION_CONTRACT> ZEROES = {0};
+
+} // namespace
+
+
+namespace multy_core
+{
+namespace internal
+{
+
+EthereumPayloadDataStream::EthereumPayloadDataStream()
+        :m_data()
+{
+    m_data.reserve(256);
+}
+
+EthereumPayloadDataStream::~EthereumPayloadDataStream()
+{
+}
+
+BinaryDataPtr EthereumPayloadDataStream::get_content() const
```

Comment:
You are creating a copy here which might be not required, like e.g. you are calling `get_content()` in `EthereumPayloadDataStream& operator<<(EthereumPayloadDataStream& stream, const EthereumContractMethodHash& value)` twice, effectively wasting time and resources for no reason. Please make that function return plain `BinaryData`, like every other data stream does.